includes/class-wc-api.php

```diff
@@ -40,6 +43,34 @@ public function __construct() {
 	}
 
 	/**
+	 * Send the response/close connection before shutdown where extra (slow)
+	 * logic, such as syncs, may be running.
+	 *
+	 * @since 3.1.2
+	 */
+	public function rest_send_reponse_before_shutdown() {
+		if ( ! defined( 'REST_REQUEST' ) || ! REST_REQUEST || headers_sent() ) {
+			return;
+		}
+
+		if ( session_id() ) {
+			session_write_close();
+		}
+
+		// This is the cleanest approuch when available and is used on nginx.
```

Comment:
Typo `approuch`. Also, it's not exclusive to `nginx`, it's part of the `php-fpm` package which can be installed as an Apache module too (it just happens that my installation doesn't include it by default). I think it's enough to say that "it isn't always present".